src/lmflow/models/hf_decoder_model.py

```diff
@@ -226,13 +227,14 @@ def __init__(
                     model_args.model_name_or_path,
                 )
 
-            self.tokenizer = AutoTokenizer.from_pretrained(model_args.model_name_or_path)
             self.backend_model_full = self.backend_model
             if peft_model_id is not None:
                 self.backend_model = PeftModel.from_pretrained(
                     self.backend_model, peft_model_id
                 )
-
+            
```

Comment:
#necessary for llama,gpt2 and other decoder model

should be 
#necessary for llama, gpt2 and other decoder model